### ISSUE TITLE
allow computing just a small number of variables

### ISSUE THREAD

**Issue by keewis** [pydata/xarray#8607]
### Is your feature request related to a problem?

I frequently find myself computing a handful of variables of a dataset (typically coordinates) and assigning them back to the dataset, and wishing we had a method / function that allowed that.

### Describe the solution you'd like

I'd imagine something like
```python
ds.compute(variables=variable_names)
```
but I'm undecided on whether that's a good idea (it might make `.compute` more complex?)

### Describe alternatives you've considered

So far I've been using something like
```python
ds.assign_coords({k: lambda ds: ds[k].compute() for k in variable_names})
ds.pipe(lambda ds: ds.merge(ds[variable_names].compute()))
```
but both are not easy to type / understand (though having `.merge` take a callable would make this much easier). Also, the first option computes variables separately, which may not be ideal?

**Comment by max-sixty** [comment]
I might well be missing something but does `ds[vars].load()` do this?

**Comment by keewis** [comment]
it does only address one part of it: I want to go from "all variables are lazy arrays" to "some variables are lazy and some are in-memory" (just as a reminder, the difference between `load` and `compute` is "in-place" vs. "copy"/"not in-place").

In other words:
```python
import xarray as xr
import numpy as np

# all variables in `ds` are lazy
ds = xr.Dataset({"a": ("x", np.arange(100)), "b": ("x", np.arange(100))}, coords={"y": ("x", np.arange(100))}).chunk({"x": 10})

ds.compute()  # no lazy variables
# ds.load()  # same result as `compute`, but `ds` got modified

ds[["a", "y"]].compute()  # no lazy variables, `b` got dropped
# ds[["a", "y"]].load()  # no lazy variables, `b` got dropped (and `ds` got modified)

# `a` and `y` are in-memory and `b` is lazy
ds.compute(variables=["a", "y"])  # the proposed syntax
ds.pipe(lambda ds: ds.merge(ds[["a", "y"]].compute()))
ds.assign_coords({k: lambda ds: ds[k].compute() for k in ["a", "y"]})
```

**Comment by dcherian** [comment]
`ds = ds.update(ds[vars].compute())` does not seem too bad to me.

**Comment by keewis** [comment]
that's true, with the downside that `update` works in-place (plus, the return value of `ds.update` is deprecated and subject to removal) and does not allow callables, so for example
```python
(
    xr.open_dataset(..., chunks={})
    .update(lambda ds: ds[vars].compute())
)
```
does not work. So it would be essentially the same thing as the `.merge` version I'm currently using, which we could extend to allow callables. Also note that only the `.assign_coords` version works for `DataArray`s (unless I'm missing something?)

(To be clear, I'm totally open to suggestions that don't make `compute` / `load` more complex)
